### PR TITLE
feat: 1991 generic async2sync

### PR DESCRIPTION
**Pull request by eoln**
- Async2SyncModel generator
- generate PartiesModel

### CODE REVIEW

File: src/test/unit/outboundApi/handlers.test.js
Comment (kleyow) [inline src/test/unit/outboundApi/handlers.test.js:502]: `.only` intentional?

File: src/lib/model/Async2SyncModel.js
Comment (kleyow) [inline src/lib/model/Async2SyncModel.js:22]: nit: spacing

**Comment by kleyow** [comment]
+1 from me, I'll let the code owners take a look at the abstraction.

File: src/test/unit/outboundApi/handlers.test.js
Comment (eoln) [reply src/test/unit/outboundApi/handlers.test.js]: thx!

File: src/lib/model/Async2SyncModel.js
Comment (lewisdaly) [inline src/lib/model/Async2SyncModel.js:81]: Do you need to pass in the `fsm` here? The function signature below takes 2 arguments.

File: src/lib/model/Async2SyncModel.js
Comment (lewisdaly) [inline src/lib/model/Async2SyncModel.js:150]: This signature doesn't match the above jsdoc

File: src/lib/model/Async2SyncModel.js
Comment (lewisdaly) [inline src/lib/model/Async2SyncModel.js:168]: This is a big number if the timeout is in seconds - maybe it's meant to be in ms?

File: src/lib/model/Async2SyncModel.js
Comment (lewisdaly) [inline src/lib/model/Async2SyncModel.js:240]: The function signature doesn't match the above function. Is `specStateMachine` global or scoped somewhere outside this maybe?

File: src/lib/model/PartiesModel.js
Comment (lewisdaly) [inline src/lib/model/PartiesModel.js:45]: jsdoc doesn't match the function's params

**Review by lewisdaly — COMMENTED** [review]
Thanks @eoln, looks mostly good to me. I've left a few questions about some functions here and there.

File: src/lib/model/Async2SyncModel.js
Comment (eoln) [reply src/lib/model/Async2SyncModel.js:168]: `.requestProcessingTimeoutSeconds` the rest of the code in sdk-scheme-adapter is using this as it would be in seconds

File: src/lib/model/Async2SyncModel.js
Comment (eoln) [reply src/lib/model/Async2SyncModel.js:81]: the function below is `onRequestAction` and it is internally invoked by FSM as a result of calling transition `requestAction` here

Comment (eoln) [reply src/lib/model/Async2SyncModel.js:81]: and FSM intentionally pass as first parameter its instance to transition notification handler <- that is the weird way how FSM working (not documented by the way)

**Review by bushjames — CHANGES_REQUESTED** [review]
@eoln Will you be making additional PRs to refactor all models to this pattern? I have a particular interest in the outbound transfers model; that state machine is a little more complex than the simple "single async to single sync" pattern expressed by your Async2SyncModel.

File: src/lib/model/Async2SyncModel.js
Comment (eoln) [reply src/lib/model/Async2SyncModel.js:240]: `specStateMachine` is defined in L48

**Comment by bushjames** [comment]
I see some of my comments appear separate from those of Lewis, some of my comments are comments on his review comments in case that is not clear.

File: src/lib/model/PartiesModel.js
Comment (eoln) [reply src/lib/model/PartiesModel.js:48]: fixed!

File: src/lib/model/Async2SyncModel.js
Comment (eoln) [reply src/lib/model/Async2SyncModel.js:148]: fixed!

File: src/lib/model/Async2SyncModel.js
Comment (eoln) [reply src/lib/model/Async2SyncModel.js:81]: > @eoln Will you be making additional PRs to refactor all models to this pattern? I have a particular interest in the outbound transfers model; that state machine is a little more complex than the simple "single async to single sync" pattern expressed by your Async2SyncModel.

it is our final goal to be able to specify/generate more complex workflows or at least use some simpler composition pattern. Up to PISP demo (April) time, we(PISP) will treat the rest of the code as technical debt, and refactor when we will be ready -> we need to collect more knowledge, and probably first we need to improve code coverage for these complex workflows before starting refactoring.

Comment (bushjames) [reply src/lib/model/Async2SyncModel.js:81]: Many thanks.

**Review by bushjames — APPROVED** [review]

File: src/lib/model/Async2SyncModel.js
Comment (lewisdaly) [reply src/lib/model/Async2SyncModel.js:166]: Yep you guys are right - I'm thinking backwards today

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**
